### PR TITLE
[Merged by Bors] - feat: the roots of a Lie algebra with non-singular Killing form span the dual of the Cartan subalgebra

### PR DESCRIPTION
**Pull request by ocfnash**
In combination with existing work the changes here mean that the following now works (and furthermore, makes no assumptions about characteristic):
```lean
example {K L : Type*} [Field K] [IsAlgClosed K] [LieRing L] [LieAlgebra K L]
    [FiniteDimensional K L] [LieAlgebra.IsKilling K L]
    (H : LieSubalgebra K L) [H.IsCartanSubalgebra] :
    Submodule.span K (range (LieModule.weight.toLinear K H L)) = ⊤ := by
  simp
```

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Dual.lean [hunk 1]
```diff
@@ -563,6 +563,32 @@ theorem forall_dual_apply_eq_zero_iff (v : V) : (∀ φ : Module.Dual K V, φ v
   rfl
 #align module.forall_dual_apply_eq_zero_iff Module.forall_dual_apply_eq_zero_iff
 
+@[simp]
+theorem subsingleton_dual_iff :
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Dual.lean:567]: Should the backwards direction also be an instance?

File: Mathlib/LinearAlgebra/Dual.lean [hunk 2]
```diff
@@ -563,6 +563,32 @@ theorem forall_dual_apply_eq_zero_iff (v : V) : (∀ φ : Module.Dual K V, φ v
...
+    Subsingleton (Dual K V) ↔ Subsingleton V := by
+  refine ⟨fun h ↦ ⟨fun v w ↦ ?_⟩, fun h ↦ ⟨fun f g ↦ ?_⟩⟩
+  · rw [← sub_eq_zero, ← forall_dual_apply_eq_zero_iff K (v - w)]
+    intros f
+    simp [Subsingleton.elim f 0]
+  · ext v
+    simp [Subsingleton.elim v 0]
+
+@[simp]
+theorem nontrivial_dual_iff :
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Dual.lean:577]: Idem.

**Review by jcommelin — APPROVED** [review]
LGTM, mod simple question.

bors d+

File: Mathlib/LinearAlgebra/Dual.lean (hunk 1 above)
Comment (ocfnash) [reply Mathlib/LinearAlgebra/Dual.lean:567]: Thanks, I had actually meant to do this.

**Pull request closed without merging**